Godeps/_workspace/src/k8s.io/kubernetes/pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -84,6 +97,122 @@ func addConversionFuncs(scheme *runtime.Scheme) {
 	}
 }
 
+func Convert_v1beta1_ReplicaSet_to_api_ReplicationController(in *ReplicaSet, out *api.ReplicationController, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*ReplicaSet))(in)
+	}
+	if err := api.Convert_unversioned_TypeMeta_To_unversioned_TypeMeta(&in.TypeMeta, &out.TypeMeta, s); err != nil {
+		return err
+	}
+	if err := v1.Convert_v1_ObjectMeta_To_api_ObjectMeta(&in.ObjectMeta, &out.ObjectMeta, s); err != nil {
+		return err
+	}
+	if err := Convert_v1beta1_ReplicaSetSpec_to_api_ReplicationControllerSpec(&in.Spec, &out.Spec, s); err != nil {
+		// TODO: Any better way to do this?
+		if !strings.Contains(err.Error(), "invalid label selector") {
+			return err
+		}
+		if out.Annotations == nil {
+			out.Annotations = make(map[string]string)
+		}
+		out.Annotations[nonConvertibleAnnotation] = ".spec.selector"
```

Comment:
> This is applicable only if we store both kinds under the same key which is what we want to do, right? So rc/frontend and rs/frontend is the same thing. Because upstream that's not the case.

It would still allow blind roundtripping through clients.  If we're fastidious about the pattern, it could allow for faithful conversions (or at least detectable failures) in the case old client/new server with different API levels.